src/pudl/convert/datapkg_to_rst.py

```diff
@@ -0,0 +1,120 @@
+"""
+Module to convert json metadata into rst files.
+
+All of the information about the transformed pudl tables, namely their fields
+types and descriptions, resides in the datapackage metadata. This module makes
+that information available to users, without duplicating any data, by converting
+json metadata files into documentation-compatible rst files. The functions
+serve to extract the field names, field data types, and field descriptions of
+each pudl table and outputs them in a manner that automatically updates the
+read-the-docs.
+
+datapky2rst_many() reads the metadata json file, extracts the afformentioned
+data, and puts it into an rst file with tags and titles for each of the unique
+tables. This way, these tags can be referenced and appear in other rst files.
+
+"""
+
+import json
+import logging
+import sys
+from pathlib import Path
+
+from jinja2 import Template
+
+logger = logging.getLogger(__name__)
+
+
+path = Path.cwd() / 'src/pudl/package_data/meta/datapkg/datapackage.json'
+metadata = open(path)
+metadata_json = json.load(metadata)
+
+
+def datapkg2rst_one():
+    """Convert json metadata to a single rst file."""
+    logger.info("Converting json metadata into an rst file")
+    template = Template(rst_template_one)
+    rendered = template.render(metadata_json)
+    # Create or overwrite an rst file containing the field descriptions of the input table
+    with open('docs/data_sources/metadata.rst', 'w') as f:
+        f.seek(0)  # Used to overwrite exisiting content
+        f.write(rendered)
+        f.truncate()  # Used to overwrite exisiting content
+
+
+def datapkg2rst_many(table):
+    """Convert json metadata to multiple rst files."""
+    logger.info("Converting json metadata into rst files")
+    template = Template(rst_template_many)
+    rendered = template.render(metadata_json, table=table).strip('\n') + ('\n')
+    # Create or overwrite an rst file containing the field descriptions of the input table
+    with open(f'docs/data_sources/{table}.rst', 'w') as f:
+        f.seek(0)  # Used to overwrite exisiting content
+        f.write(rendered)
+        f.truncate()  # Used to overwrite exisiting content
+
+
+def main():
```

Comment:
It would be good to add at least a minimal command line parsing function -- every command line utility should respond to at least the `-h` and `--help` flags with some explanation of its existence and usage, even if it doesn't have real "options" for the user to control its behavior.

This script is a perfect candidate for the classic "reads from standard input and writes to standard output" usage since it consumes a chunk of text (the JSON) and generates another chunk of text (the rST), maybe with the option of specifying input and output files explicitly with e.g. `-i` / `--input` and `-o` / `--output`. Allowing the input source and output destination to be controlled at runtime would make it more flexible / testable, and avoid the need to hardcode that information as constants inside the script.

Here's a good [guide to command line interface design](https://clig.dev/).

Reply:
> I pulled the branch down and ran the docs build to see what it looks like in the flesh, and wonder if there's some way we can allow line breaks in the description column -- having to scroll to see the whole width of the field isn't idea.
> 
> ![image](https://user-images.githubusercontent.com/596279/106839157-40ce7600-6663-11eb-84cd-67b86deacb46.png)

Unfortunately this does not have a quick fix that I'm aware of